File: app/src/main/java/sakaitakao/android/permissionsearch/entity/EasySearchInfo.java
package sakaitakao.android.permissionsearch.entity;

import java.io.Serializable;
import java.util.List;

/**
 * パーミッションとパーミッションを保持するアプリのリスト
 * 
 * @author takao
 * 
 */
public class EasySearchInfo implements Serializable {

	private static final long serialVersionUID = 1L;

	/**
	 * 項目名
	 */
	public String name;

	/**
	 * 説明
	 */
	public String description;

	/**
	 * 権限名のパターン
	 */
	public List<String> permissionNamePatternList;

	/**
	 * プロテクションレヴェル
	 */
	public List<Integer> protectionLevelList;

	/**
	 * コンストラクタ
	 */
	public EasySearchInfo() {
	}
}

File: app/src/main/kotlin/sakaitakao/android/permissionsearch/entity/EasySearchInfo.kt
package sakaitakao.android.permissionsearch.entity

import java.io.Serializable
import java.util.*

/**
 * パーミッションとパーミッションを保持するアプリのリスト

 * @author takao
 */
class EasySearchInfo : Serializable {

    /**
     * 項目名
     */
    public var name: String = ""

    /**
     * 説明
     */
    public var description: String = ""

    /**
     * 権限名のパターン
     */
    public var permissionNamePatternList: List<String> = ArrayList()

    /**
     * プロテクションレヴェル
     */
    public var protectionLevelList: List<Int> = ArrayList()


    companion object {
        private val serialVersionUID = 1L
    }
}
